commonmark/src/main/java/org/commonmark/parser/Parser.java

```diff
@@ -116,6 +118,37 @@ public Builder extensions(Iterable<? extends Extension> extensions) {
         }
 
         /**
+         * Describe the list of markdown features the parser will recognize and parse.
+         *
+         * By default, Commonmark will recognize and parse the following set of core markdown features:
+         *
+         * Heading (#) - Heading.class
+         * HTML (<html></html>) - HtmlBlock.class
+         * Horizontal Rule / Thematic Break (---) - ThematicBreak.class
+         * Fenced Code Block (```) - FencedCodeBlock.class
+         * Indented Code Block - IndentedCodeBlock.class
+         * Block Quote (>) - BlockQuote.class
+         * Ordered / Unordered List (>) - ListBlock.class
+         *
+         * To parse only a subset of the features listed above, pass a list of each feature's associated Node class.
+         * E.g., to only parse Headings and Lists:
+         * <pre>
+         *     {@code
+         *     Parser.builder().allowedBlockTypes(Heading.class, ListBlock.class);
+         *     }
+         * </pre>
+         *
+         * @param allowedBlockTypes A list of nodes the parser will parse.
+         *                     If this list is empty, the parser will not recognize any Commonmark core markdown features.
+         *
+         * @return {@code this}
+         */
+        public Builder allowedBlockTypes(List<Class<? extends Block>> allowedBlockTypes) {
```

Comment:
Wondering if it should be a `Set` instead of a `List`, seems to fit the use better.